import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/backupuploader/BackupUploadService.java

```diff
@@ -100,25 +110,36 @@ private void scheduleUploadWithDelay(IDicomWebClient webClient, BackupState back
         throw new IBackupUploader.BackupException(eex.getCause());
       }
     } else {
-      throw getNoResendAttemptLeftException(uniqueFileName);
+      throw getNoResendAttemptLeftException(null, uniqueFileName);
     }
   }
 
   private void scheduleUploadWithDelayExceptionally(IDicomWebClient webClient, BackupState backupState) {
     try {
       scheduleUploadWithDelay(webClient, backupState);
-      } catch (IBackupUploader.BackupException ex) {
+    } catch (IBackupUploader.BackupException ex) {
       throw new CompletionException(ex);
     }
   }
 
-  private void throwOnNoResendAttemptsLeft(String uniqueFileName) throws CompletionException {
-    throw new CompletionException(getNoResendAttemptLeftException(uniqueFileName));
+  private void throwOnNoResendAttemptsLeft(int dicomStatus, String uniqueFileName) throws CompletionException {
+    throw new CompletionException(getNoResendAttemptLeftException(dicomStatus, uniqueFileName));
+  }
+
+  private void throwOnHttpFilterFail(int dicomStatus, String uniqueFileName) throws CompletionException {
+    String errorMessage = "sopInstanceUID=" + uniqueFileName + ". Filtered by httpCode.";
```

Comment:
Can we reword this slightly to say "Not retried due to HTTP code" to make it more clear?